src/arch/simddetect.cpp

```diff
@@ -55,10 +55,12 @@
 #if defined(HAVE_NEON) && !defined(__aarch64__)
 #  ifdef ANDROID
 #    include <cpu-features.h>
-#  else
-/* Assume linux */
+#  elif defined(__linux__)
 #    include <asm/hwcap.h>
 #    include <sys/auxv.h>
+#  elif defined(__FreeBSD__)
```

Comment:
I'd prefer to check for Android first, then FreeBSD, then the rest (no explicit check for Linux). That's closer to the current code and perhaps covers more cases.

Reply:
If you do it that way, the build will fail on non-Linux operating systems.  I have specifically designed this patch to permit compilation on operating systems for which we don't know how to obtain the ELF aux vectors, falling back to the safe assumption that NEON is not supported.

Reply:
I don't think that the code should silently assume that NEON is unsupported because that might result in code which is slower than necessary. Failing seems better to me, because it allows investigating and fixing the problem.

Reply:
As we are talking about 32 bit ARM systems (which typically are not very fast), using NEON is especially important to get at least an acceptable performance.

Reply:
Ideally, a configure test would be used to check which way of finding NEON support is available.  Just failing the build seems nasty to users of other operating systems.  I mean I can change the patch like that if you insist.

Reply:
Yes, a configure test (which only runs if necessary) would be the best solution.

Reply:
I've never designed a configure test before.  I'll try to figure it out and report back.

Reply:
I have added configure tests as directed.  Please also check and revise the `AC_CONDITIONAL` usage.  I don't think these are meant to be redefined.